src/modules/LimitOrders.ts

```diff
@@ -0,0 +1,353 @@
+import {
+  address,
+  ContractCallOptions,
+  ContractConstantCallOptions,
+  Integer,
+  LimitOrder,
+} from '../../src/types';
+
+const EIP712_ORDER_STRUCT = [
+  { type: 'uint256', name: 'makerMarket' },
+  { type: 'uint256', name: 'takerMarket' },
+  { type: 'uint256', name: 'makerAmount' },
+  { type: 'uint256', name: 'takerAmount' },
+  { type: 'address', name: 'makerAccountOwner' },
+  { type: 'uint256', name: 'makerAccountNumber' },
+  { type: 'address', name: 'takerAccountOwner' },
+  { type: 'uint256', name: 'takerAccountNumber' },
+  { type: 'uint256', name: 'expiration' },
+  { type: 'uint256', name: 'salt' },
+];
+
+const EIP712_ORDER_STRUCT_STRING =
+  'LimitOrder(' +
+  'uint256 makerMarket,' +
+  'uint256 takerMarket,' +
+  'uint256 makerAmount,' +
+  'uint256 takerAmount,' +
+  'address makerAccountOwner,' +
+  'uint256 makerAccountNumber,' +
+  'address takerAccountOwner,' +
+  'uint256 takerAccountNumber,' +
+  'uint256 expiration,' +
+  'uint256 salt' +
+  ')';
+
+export class LimitOrders {
+  private contracts: Contracts;
+  private web3: Web3;
+  private networkId: number;
+
+  // ============ Constructor ============
+
+  constructor(
+    contracts: Contracts,
+    web3: Web3,
+    networkId: number,
+  ) {
+    this.contracts = contracts;
+    this.web3 = web3;
+    this.networkId = networkId;
+  }
+
+  // ============ Sender Contract Methods ============
+
+  public async approveOrder(
+    order: LimitOrder,
+    options?: ContractCallOptions,
+  ): Promise<any> {
+    const orderHash = this.getOrderHash(order);
+    return await this.contracts.callContractFunction(
+      this.contracts.limitOrders.methods.approveOrder(orderHash),
+      options,
+    );
+  }
+
+  public async cancelOrder(
+    order: LimitOrder,
+    options?: ContractCallOptions,
+  ): Promise<any> {
+    const orderHash = this.getOrderHash(order);
+    return await this.contracts.callContractFunction(
+      this.contracts.limitOrders.methods.cancelOrder(orderHash),
+      options,
+    );
+  }
+
+  // ============ Getter Contract Methods ============
+
+  public async getOrderStatus(
+    order: LimitOrder,
+    options?: ContractConstantCallOptions,
+  ): Promise<{status: number, makerFilledAmount: Integer}> {
+    const orderHash = this.getOrderHash(order);
+
+    const [
+      status,
+      makerFilledAmount,
+    ] = await Promise.all([
+      this.contracts.callConstantContractFunction(
+        this.contracts.limitOrders.methods.g_status(
+          order.makerAccountOwner,
+          orderHash,
+        ),
+        options,
+      ),
+      this.contracts.callConstantContractFunction(
+        this.contracts.limitOrders.methods.g_filled(
+          orderHash,
+        ),
+        options,
+      ),
+    ]);
+
+    return {
+      status: parseInt(status, 10),
+      makerFilledAmount: new BigNumber(makerFilledAmount),
+    };
+  }
+
+  // ============ Signing Methods ============
+
+  public async ethSignTypedOrderV3(
```

Comment:
what is the difference between V3 and the normal one? can you put some comments?